### DISCUSSION TITLE
New syntax for table caption

### DISCUSSION THREAD

**Discussion by raphaeljoie** [remarkjs/remark#804] [Ideas] [answered]
I'm using Remark via MDX for NextJS, and using custom components for rendering.

# The hack with image
To parametrise the images display I use the standard "alt" attribute

```md
![{align: 'center'}](image.png)
```
and later
```jsx
export default function Image({src, alt}) {
    const options = JSON.parse(alt)

    return <img src={src} style={{textAlign: options.align}} />
}
```

But the table doesn't provide such a built-in-way to push a caption in the metadata of the table

# The solution with tables
What about an extension of the std table definition, like
```md
| a | b |
| --- | --- |
| 1 | 2 |
[caption]
```

**Comment by ChristianMurphy** [comment]
Markdown as a semi-standard way to add/wrap custom content types, directives.
https://github.com/remarkjs/remark-directive
It may make sense to wrap the table in a caption directive.

**Reply by raphaeljoie** [reply]
Hey @ChristianMurphy, thank you for your reply, and also thanks a lot for the time dedicated to this opensourced project.

I know that remark-directive can do a lot, but the reason I came here is because the doc invites to publish new ideas for plugins. Redirecting me to the writing of my own directive is a valid solution. But I believe your interest here is more to capture similar needs to feed a pipeline of potential plug-and-play plugins, rather than inviting people to write their own.

You can close the thread and/or accept your response if you believe that I'm wrong and confirm that writing my own directive is the **best** answer. Otherwise it can remains open and eventually capture the attention of others also looking for a no-code solution to caption tables :)

Thanks again for the framework and the time dedicated.

**Reply by ChristianMurphy** [reply]
Again there are some existing solutions:
* https://github.com/zestedesavoir/zmarkdown/tree/master/packages/remark-captions
* https://github.com/remarkjs/remark-directive

If you don't like the syntax of either
https://github.com/micromark/micromark#extending-markdown has a write up on how syntax extensions can be created.

> I believe your interest here is more to capture similar needs to feed a pipeline of potential plug-and-play plugins, rather than inviting people to write their own.

**It is**: a place to discuss solutions, get advise on syntax, and ideas for alternatives.
**It is not**: make a suggestion and others will write it for you.

> if you believe that I'm wrong and confirm that writing my own directive is the **best** answer

**best** is highly subjective.
From how you framed your idea and from how you framed your follow up response. Your goal would appear to be a low code solution, which others have written and maintain.
Creating a new syntax extension is _not that_, it requires a fair amount of code and maintenance.
Using `remark-captions` or `remark-directive` would be a more low code solution.

**Reply by raphaeljoie** [reply]
https://github.com/zestedesavoir/zmarkdown/tree/master/packages/remark-captions is the answer I was looking for. Thanks :)

**Comment by raphaeljoie** [comment, marked as answer]
Thank you @ChristianMurphy for pointing at the existing package

https://github.com/zestedesavoir/zmarkdown/tree/master/packages/remark-captions
